Classes/ContentReleaseManager.php

```diff
@@ -43,24 +46,42 @@ class ContentReleaseManager
     const NO_PREVIOUS_RELEASE = 'NO_PREVIOUS_RELEASE';
 
-    public function startIncrementalContentRelease()
+    public function startIncrementalContentRelease(string $currentContentReleaseId = null, Workspace $workspace = null, array $additionalVariables = []): ContentReleaseIdentifier
     {
         $redis = $this->redisClientManager->getPrimaryRedis();
-        $currentContentReleaseId = $redis->get(self::REDIS_CURRENT_RELEASE_KEY);
+        if ($currentContentReleaseId) {
+            $currentContentReleaseId = $redis->get(self::REDIS_CURRENT_RELEASE_KEY);
+        }
 
+        $contentReleaseId = ContentReleaseIdentifier::create();
         // the currentContentReleaseId is not used in any pipeline step in this package, but is a common need in other
         // use cases in extensions, e.g. calculating the differences between current and new release
-        $this->prunnerApiService->schedulePipeline(PipelineName::create('do_content_release'), ['contentReleaseId' => (string)time(), 'currentContentReleaseId' => $currentContentReleaseId ?: self::NO_PREVIOUS_RELEASE, 'validate' => true]);
+        $this->prunnerApiService->schedulePipeline(PipelineName::create('do_content_release'), array_merge([
+            'contentReleaseId' => $contentReleaseId,
+            'currentContentReleaseId' => $currentContentReleaseId ?: self::NO_PREVIOUS_RELEASE,
+            'validate' => true,
+            'workspaceName' => $workspace ? $workspace->getName() : 'live',
+        ], $additionalVariables));
+        return $contentReleaseId;
     }
 
     // the validate parameter can be used to intentionally skip the validation step for this release
-    public function startFullContentRelease(bool $validate = true)
+    public function startFullContentRelease(bool $validate = true, string $currentContentReleaseId = null, Workspace $workspace = null, array $additionalVariables = []): ContentReleaseIdentifier
     {
         $redis = $this->redisClientManager->getPrimaryRedis();
-        $currentContentReleaseId = $redis->get(self::REDIS_CURRENT_RELEASE_KEY);
+        if ($currentContentReleaseId) {
+            $currentContentReleaseId = $redis->get(self::REDIS_CURRENT_RELEASE_KEY);
+        }
 
+        $contentReleaseId = ContentReleaseIdentifier::create();
         $this->contentCache->flush();
-        $this->prunnerApiService->schedulePipeline(PipelineName::create('do_content_release'), ['contentReleaseId' => (string)time(), 'currentContentReleaseId' => $currentContentReleaseId ?: self::NO_PREVIOUS_RELEASE, 'validate' => $validate]);
+        $this->prunnerApiService->schedulePipeline(PipelineName::create('do_content_release'), array_merge([
+            'contentReleaseId' => $contentReleaseId,
+            'currentContentReleaseId' => $currentContentReleaseId ?: self::NO_PREVIOUS_RELEASE,
+            'validate' => $validate,
+            'workspaceName' => $workspace ? $workspace->getName() : 'live',
+        ], $additionalVariables));
```

Comment:
can we turn the order around? (merge $additionalVariables with the fixed values); to ensure an additionalVariable can NEVER override one of the crucial properties?)